fbw-a380x/src/wasm/fadec_a380x/src/Fadec/FadecSimData_A380X.hpp

```diff
@@ -0,0 +1,461 @@
+  NamedVariablePtr refuelRate;
+  NamedVariablePtr refuelStartedByUser;
+  NamedVariablePtr startState;
+  NamedVariablePtr thrustLimitClimb;
+  NamedVariablePtr thrustLimitFlex;
+  NamedVariablePtr thrustLimitIdle;
+  NamedVariablePtr thrustLimitMct;
+  NamedVariablePtr thrustLimitToga;
+  NamedVariablePtr thrustLimitType;
+  NamedVariablePtr wingAntiIce;
+
+  // ===============================================================================================
+
+  /**
+   * @brief Initializes the FadecSimData_A380X object.
+   * @param dm Pointer to the DataManager object that is used to manage the data definitions and variables.
+   */
+  void initialize(DataManager* dm) {
+    initDataDefinitions(dm);
+    initEvents(dm);
+    initSimvars(dm);
+    initLvars(dm);
+    LOG_INFO("Fadec::FadecSimData_A32NX initialized");
+  }
+
+  void initDataDefinitions(DataManager* dm) {
+    atcIdDataPtr = dm->make_datadefinition_var<AtcIdData>("ATC ID DATA", atcIdDataDef, NO_AUTO_UPDATE);
+
+    fuelFeedTankDataPtr = dm->make_datadefinition_var<FuelFeedTankData>("FUEL FEED TANK DATA", fuelFeedTankDataDef);
+    fuelFeedTankDataPtr->requestPeriodicDataFromSim(SIMCONNECT_PERIOD_VISUAL_FRAME);
+
+    fuelTankDataPtr = dm->make_datadefinition_var<FuelTankData>("FUEL TANK DATA", fuelTankDataDef);
+    fuelTankDataPtr->requestPeriodicDataFromSim(SIMCONNECT_PERIOD_VISUAL_FRAME);
+
+    oilTempDataPtr[E1] = dm->make_datadefinition_var<OilTempData>("GENERAL ENG OIL TEMPERATURE 1", oilTempE1DataDef, NO_AUTO_UPDATE);
+    oilTempDataPtr[E2] = dm->make_datadefinition_var<OilTempData>("GENERAL ENG OIL TEMPERATURE 2", oilTempE2DataDef, NO_AUTO_UPDATE);
+    oilTempDataPtr[E3] = dm->make_datadefinition_var<OilTempData>("GENERAL ENG OIL TEMPERATURE 3", oilTempE3DataDef, NO_AUTO_UPDATE);
+    oilTempDataPtr[E4] = dm->make_datadefinition_var<OilTempData>("GENERAL ENG OIL TEMPERATURE 4", oilTempE4DataDef, NO_AUTO_UPDATE);
+
+    oilPsiDataPtr[E1] = dm->make_datadefinition_var<OilPsiData>("GENERAL ENG OIL PRESSURE 1", oilPsiE1DataDef, NO_AUTO_UPDATE);
+    oilPsiDataPtr[E2] = dm->make_datadefinition_var<OilPsiData>("GENERAL ENG OIL PRESSURE 2", oilPsiE2DataDef, NO_AUTO_UPDATE);
+    oilPsiDataPtr[E3] = dm->make_datadefinition_var<OilPsiData>("GENERAL ENG OIL PRESSURE 3", oilPsiE3DataDef, NO_AUTO_UPDATE);
+    oilPsiDataPtr[E4] = dm->make_datadefinition_var<OilPsiData>("GENERAL ENG OIL PRESSURE 4", oilPsiE4DataDef, NO_AUTO_UPDATE);
+
+    engineCorrectedN3DataPtr[E1] = dm->make_datadefinition_var<CorrectedN3Data>("TURB ENG CN2 1", engine1CN3DataDef, NO_AUTO_UPDATE);
+    engineCorrectedN3DataPtr[E2] = dm->make_datadefinition_var<CorrectedN3Data>("TURB ENG CN2 2", engine2CN3DataDef, NO_AUTO_UPDATE);
+    engineCorrectedN3DataPtr[E3] = dm->make_datadefinition_var<CorrectedN3Data>("TURB ENG CN2 3", engine3CN3DataDef, NO_AUTO_UPDATE);
+    engineCorrectedN3DataPtr[E4] = dm->make_datadefinition_var<CorrectedN3Data>("TURB ENG CN2 4", engine4CN3DataDef, NO_AUTO_UPDATE);
```

Comment:
Even though this is already covered by a comment to the struct above

 > Note: the sim does not have a direct N3 value, so we use N2 as a proxy

I still think there should be a comment/note above here which again notes and refers to this, as it is important.